### ISSUE TITLE
Animal sniffer doesn't handle change in `ByteBuffer.flip` method for Java 9

### ISSUE THREAD

**Issue by wjam** [mojohaus/animal-sniffer#4]
While testing building various open source projects against the latest build of the Java 9, it appears that animal sniffer doesn't like the change to the `Buffer.flip` method on the `Buffer` subclasses and instead reports the following error (these errors were reported while building Maven):
```
[ERROR] .../maven/maven-compat/src/main/java/org/apache/maven/repository/legacy/DefaultUpdateCheckManager.java:253: Undefined reference: java.nio.ByteBuffer java.nio.ByteBuffer.flip()
[ERROR] .../maven/maven-compat/src/main/java/org/apache/maven/repository/legacy/DefaultUpdateCheckManager.java:278: Undefined reference: java.nio.ByteBuffer java.nio.ByteBuffer.flip()
```

The details of the changes to the API is that the on all subclasses of `Buffer`, the `flip` method will return the subclass rather than `Buffer` (so `ByteBuffer.flip` will return `ByteBuffer` rather than `Buffer`)

**Comment by rfscholte** [comment]
Is already fixed with MANIMALSNIFFER-49, which is available in the latest released version. There's also an IT confirming this.

**Comment by onacit** [comment]
@wjam Would you please how you've solved? Especially with the animal-sniffer-maven-plugin? Thanks.

**Comment by balrok** [comment]
@onacit I found with the maven plugin in version 1.20 it is not throwing an error - but just a warning, while all later versions will report this as an error

**Comment by onacit** [comment]
@balrok Confirmed.

```xml
<maven.compiler.release>8</maven.compiler.release>
```

Thank you.
